stdlib/priorityqueue.gr

```diff
@@ -0,0 +1,233 @@
+/**
+ * @module PriorityQueue: A mutable priority queue implementation. A priority queue is a data structure that maintains elements in a priority order. Elements with higher priority are served before elements with lower priority when extracting from the priority queue.
+ *
+ * @example import PriorityQueue from "priorityqueue"
+ *
+ * @since v0.5.3
+ */
+
+import Array from "array"
+import List from "list"
+import Number from "number"
+import Option from "option"
+
+/**
+ * @section Types: Type declarations included in the PriorityQueue module.
+ */
+
+/**
+ * Mutable data structure which maintains a priority order for its elements.
+ */
+record PriorityQueue<a> {
+  mut size: Number,
+  mut array: Array<Option<a>>,
+  comp: (a, a) -> Number,
+}
+
+/**
+ * @section Values: Functions for working with PriorityQueues.
+ */
+
+let swap = (i1, i2, array) => {
+  let t = array[i2]
+  array[i2] = array[i1]
+  array[i1] = t
+}
+
+let get = (array, i) => Option.unwrap(array[i])
```

Comment:
```suggestion
let get = (array, i) => Option.unwrap(array[i])
```
Should we use `Option.expect` here so it would actually fail with a better error message if it was somehow `None`?

Reply:
In theory this shouldn't happen but I guess that would be a good idea in case there is a bug somewhere

Reply:
In the compiler, we often do something like `"impossible by SOMETHING_UNIQUE"` so we can track down those bugs. It looks like regex.gr has a few things like `"impossible (mergeAdjacent)"` and `"Impossible: continue_m"`